pkg/cloud/services/interfaces.go

```diff
@@ -18,6 +18,7 @@ package services
 
 import (
 	apimachinerytypes "k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
```

Comment:
This is similar to the other comment I left you on a different PR :smile: I think we need to be careful with the dependencies between packages or everything will be tangled up. Can we maybe return a custom error that we can check in the caller somehow, so that we don't have to make the `services` package depend on `controller-runtime`?

Reply:
For this one, I wouldn't make the effort to pull it out. Other code below this package use the `Result` object as well, and also have other dependencies on `controller-runtime`.